Com esse backport eu tambem mudei o container de CI para Python 3.7 de volta. O Python 3.6 ta End of Life ja faz uns 2 anos e  de qualquer forma, antes de usar o Gituhub Actions, quando a gente usava o Travis ainda a gente ja testava a localização com o Python 3.7. Na Akretion a gente tb roda os clientes da v12 no Python 3.7 faz tempo.

Acontece que com Python 3.6 tinha um erro na inutilização da NFe por conta do metodo fromisoformat do datetime.datetime que [so chegou com Python 3.7](https://stackoverflow.com/a/60266709)

```
2023-06-17 20:22:15,242 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: `   File "/__w/l10n-brazil/l10n-brazil/l10n_br_nfe/models/document.py", line 842, in atualiza_status_nfe 
2023-06-17 20:22:15,242 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: `     datetime.fromisoformat(infProt.dhRecbto) 
2023-06-17 20:22:15,242 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: ` AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat' 
2023-06-17 20:22:15,242 366 INFO odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: ====================================================================== 
2023-06-17 20:22:15,242 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: ERROR: test_inutilizar (odoo.addons.l10n_br_nfe.tests.test_nfe_webservices.TestNFeWebservices) 
2023-06-17 20:22:15,242 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: Traceback (most recent call last): 
2023-06-17 20:22:15,242 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: `   File "/usr/lib/python3.6/unittest/mock.py", line 1183, in patched 
2023-06-17 20:22:15,242 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: `     return func(*args, **keywargs) 
2023-06-17 20:22:15,242 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: `   File "/__w/l10n-brazil/l10n-brazil/l10n_br_nfe/tests/test_nfe_webservices.py", line 185, in test_inutilizar 
2023-06-17 20:22:15,242 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: `     inutilizar_wizard.doit() 
2023-06-17 20:22:15,242 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: `   File "/__w/l10n-brazil/l10n-brazil/l10n_br_fiscal/wizards/invalidate_number_wizard.py", line 28, in doit 
2023-06-17 20:22:15,242 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: `     wizard.do_invalidate() 
2023-06-17 20:22:15,242 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: `   File "/__w/l10n-brazil/l10n-brazil/l10n_br_fiscal/wizards/invalidate_number_wizard.py", line 24, in do_invalidate 
2023-06-17 20:22:15,242 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: `     invalidate._invalidate(self.document_id) 
2023-06-17 20:22:15,243 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: `   File "/__w/l10n-brazil/l10n-brazil/l10n_br_nfe/models/invalidate_number.py", line 85, in _invalidate 
2023-06-17 20:22:15,243 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: `     datetime.fromisoformat(date_response) 
2023-06-17 20:22:15,243 366 ERROR odoo odoo.addons.l10n_br_nfe.tests.test_nfe_webservices: ` AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat' 
```

--- a/l10n_br_nfe/models/invalidate_number.py
+++ b/l10n_br_nfe/models/invalidate_number.py
@@ -2,6 +2,7 @@
 # License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
 
 import logging
+import sys
 from datetime import datetime
 
 from erpbrasil.assinatura import certificado as cert
@@ -20,6 +21,11 @@ try:
     from erpbrasil.base.misc import punctuation_rm
 except ImportError:
     _logger.error("Biblioteca erpbrasil.base não instalada")
+
+if sys.version_info < (3, 7):
+    from backports.datetime_fromisoformat import MonkeyPatch
+
+    MonkeyPatch.patch_fromisoformat()
 
 
 class InvalidateNumber(models.Model):

--- a/test-requirements.txt
+++ b/test-requirements.txt
@@ -1,2 +1,3 @@
 vcrpy  # Needed by payment_pagseguro
 odoo-test-helper  # Needed by spec_driven_model
+backports-datetime-fromisoformat

